src/errors/validation_exception.rs

```diff
@@ -24,27 +24,60 @@ use super::types::{ErrorMode, ErrorType};
 use super::value_exception::PydanticCustomError;
 use super::ValError;
 
+#[derive(Clone)]
+#[cfg_attr(debug_assertions, derive(Debug))]
+enum WrappedError {
+    LineError(PyLineError),
+    ValidationError(ValidationError),
+}
+
+impl From<PyLineError> for WrappedError {
+    fn from(value: PyLineError) -> Self {
+        WrappedError::LineError(value)
+    }
+}
+
+impl From<ValidationError> for WrappedError {
+    fn from(value: ValidationError) -> Self {
+        WrappedError::ValidationError(value)
+    }
+}
+
+impl<'source> FromPyObject<'source> for WrappedError {
+    fn extract(ob: &'source PyAny) -> PyResult<Self> {
+        if let Ok(v) = ob.extract::<ValidationError>() {
+            Ok(WrappedError::ValidationError(v))
+        } else {
+            Ok(WrappedError::LineError(PyLineError::try_from(ob)?))
+        }
+    }
+}
+
 #[pyclass(subclass, extends=PyValueError, module="pydantic_core._pydantic_core")]
 #[derive(Clone)]
 #[cfg_attr(debug_assertions, derive(Debug))]
 pub struct ValidationError {
-    line_errors: Vec<PyLineError>,
+    exceptions: Vec<WrappedError>,
```

Comment:
can we call this `error`? I think it would make more sense and be a little clearer.